File: LeetCode/Easy/0205-isomorphic-strings/0205-isomorphic-strings.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        Span<char> sMap = stackalloc char[128];
        Span<char> tMap = stackalloc char[128];
        for (var i = 0; i < s.Length; i++)
        {
            var si = s[i];
            var ti = t[i];
            if (tMap[ti] == '\0' && sMap[si] == '\0')
            {
                tMap[ti] = si;
                sMap[si] = ti;
            }
            else if (sMap[si] != ti || tMap[ti] != si)
            {
                return false;
            }
        }
        
        return true;
    }
}

File: LeetCode/Easy/0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s_to_t_map = {}
        t_to_s_exist = set()

        for (s_char, t_char) in zip(s, t):
            if s_char in s_to_t_map:
                if s_to_t_map[s_char] != t_char:
                    return False
            else:
                if t_char in t_to_s_exist:
                    return False
                s_to_t_map[s_char] = t_char
                t_to_s_exist.add(t_char)

        return True